### ISSUE TITLE
[BUG] Cannot predict on unseen data (`XGBoost#predict` requires target column)

### ISSUE THREAD

**Issue by radekosmulski** [NVIDIA-Merlin/models#599]
### Bug description

After training a model, one might want to predict on unseen data. This is currently impossible as `predict` requires the target column.

![image](https://user-images.githubusercontent.com/2444926/180159584-34a53183-fdaf-4c88-a11e-4dd55aeb2181.png)

The workaround is to add a dummy target column, but that is not what should be needed. Also, the issue with that workaround is that a user can never be sure to what extent the black box model utilizes information contained in the target column for making predictions.

`predict` should not require access to ground truth. 

### Steps/Code to reproduce bug
```
from merlin.core.utils import Distributed
from merlin.models.xgb import XGBoost
import sklearn.datasets
import cudf
import numpy as np
import nvtabular as nvt

rows = 200
num_features = 16
X, y = sklearn.datasets.make_regression(
    n_samples=rows,
    n_features=num_features,
    n_informative=num_features // 3,
    random_state=0,
)

feature_names = [str(i) for i in range(num_features)]
df = cudf.DataFrame(np.hstack((X, y.reshape(-1, 1))), columns=feature_names+['target'], dtype=np.float32)
train_ds = nvt.Dataset(df.iloc[:160])
valid_ds = nvt.Dataset(df.iloc[40:])

with Distributed():
    model = XGBoost(schema=train_ds.schema, target_columns='target')
    model.fit(
        train_ds,
        evals=[(valid_ds, 'validation_set'),]
    )

test_ds = nvt.Dataset(df.iloc[40:].drop(columns='target'))

with Distributed():
    model.predict(test_ds)
```

### Expected behavior
`predict` shouldn't require the target column to be present. Probably nice behavior for the user would be for predict to disregard the `target` column if it is passed in along with the dataset (for instance, this could happen if a user wants to predict on the train or validation set, which is a common scenario in the case of error analysis or calculating additional metrics)
